### PR TITLE
Refuse connection from ips that are already in processing queue

### PR DESCRIPTION
**Pull request by zilm13**
@mkalinin please, take a look

### CODE REVIEW

File: ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java [hunk 1]
```diff
@@ -115,6 +92,43 @@ public void initChannel(NioSocketChannel ch) throws Exception {
             logger.error("Unexpected error: ", e);
         }
     }
+    
+    private boolean shouldRefuseIncoming(NioSocketChannel ch) {
+        if(!isInbound()) return false;
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java]: it's better to check it outside

File: ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java [hunk 2]
```diff
@@ -115,6 +92,43 @@ public void initChannel(NioSocketChannel ch) throws Exception {
             logger.error("Unexpected error: ", e);
         }
     }
+    
+    private boolean shouldRefuseIncoming(NioSocketChannel ch) {
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java]: Maybe something like `isEligibleForConnection`?

File: ethereumj-core/src/main/java/org/ethereum/net/server/ChannelManager.java [hunk 3]
```diff
@@ -200,6 +200,21 @@ public void disconnect(Channel peer, ReasonCode reason) {
         recentlyDisconnected.put(peer.getInetSocketAddress().getAddress(), new Date());
     }
 
+    /**
+     * Whether peer with the same ip is in newPeers, waiting for processing
+     * @param peerAddr      Peer address
+     * @return true if we already have connection from this address, otherwise false
+     */
+    public boolean ipAlreadyWaiting(InetAddress peerAddr) {
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/net/server/ChannelManager.java]: `isAddressInQueue`?

File: ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java (hunk 2 above)
Comment (zilm13) [reply ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java]: i like it!

Comment (zilm13) [reply ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java]: almost )

**Pull request merged**